plugins/newsdownloader.koplugin/main.lua

```diff
@@ -0,0 +1,185 @@
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local InfoMessage = require("ui/widget/infomessage")
+local UIManager = require("ui/uimanager")
+local DataStorage = require("datastorage")
+local FFIUtil = require("ffi/util")
+local util = require("frontend/util")
+local T = FFIUtil.template
+local _ = require("gettext")
+local logger = require("logger")
+local ffi = require("ffi")
+local http = require("socket.http")
+local ltn12 = require("ltn12")
+
+
+local NewsDownloader = WidgetContainer:new{}
+
+local initialized = false  -- for only once lazy initialization
+local FEED_CONFIG_FILE = "feeds.xml"
+local FILE_EXTENSION = ".html"
+local NEWS_DL_DIR_NAME = "news"
+local NEWS_DL_DIR, FEED_CONFIG_PATH
+
+local function deserializeXMLString(xml_str)
+    -- uses LuaXML https://github.com/manoelcampos/LuaXML
+    -- The MIT License (MIT)
+    -- Copyright (c) 2016 Manoel Campos da Silva Filho
+    local treehdl = require("lib/handler")
+    local libxml = require("lib/xml")
+
+    --Instantiate the object the states the XML file as a Lua table
+    local xmlhandler = treehdl.simpleTreeHandler()
+    --Instantiate the object that parses the XML to a Lua table
+    libxml.xmlParser(xmlhandler):parse(xml_str)
+    return xmlhandler.root
+end
+
+local function deserializeXML(filename)
+    logger.dbg("NewsDownloader: File to deserialize: ", filename)
+    local f, e = io.open(filename, "r")
+    if f then
+        local xmltext = f:read("*a")
+        f:close()
+        return deserializeXMLString(xmltext)
+    else
+        logger.warn("NewsDownloader: XML file not found", filename, e)
+    end
+end
+
+function NewsDownloader:init()
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function NewsDownloader:addToMainMenu(menu_items)
+    if not initialized then
+        NEWS_DL_DIR = string.format("%s/%s/", DataStorage:getDataDir(), NEWS_DL_DIR_NAME)
+        if not lfs.attributes(NEWS_DL_DIR, "mode") then
+            lfs.mkdir(NEWS_DL_DIR)
+        end
+
+        FEED_CONFIG_PATH = NEWS_DL_DIR .. FEED_CONFIG_FILE
+        if not lfs.attributes(FEED_CONFIG_PATH, "mode") then
+            logger.dbg("NewsDownloader: Creating init configuration")
+            FFIUtil.copyFile(FFIUtil.joinPath(self.path, FEED_CONFIG_FILE),
+                             FEED_CONFIG_PATH)
+        end
+        initialized = true
+    end
+
+    menu_items.rss_news_downloader = {
+        text = _("Simple News(RSS/Atom) Downloader"),
```

Comment:
"Newsfeed" seems odd to me. 
If I'd to choose between this and mine I'd choose current name. 

I can remove the atom from name, if needed, but I already have idea howto add it. 

I just don't want to make that PR any bigger (anyway it's probably one of the biggest /longest working) 
So I'll want to add that after this one is finished. 

Anyway it can be also changed later on, I think choosing the best name should block this PR..